src/proc.rs

```diff
@@ -1066,6 +1067,12 @@ impl Job {
                     parser.libdata_mut().status_count += 1;
                 }
             }
+
+            // Prevent kernel-buffered output from being written to the terminal after the prompt.
+            // We don't need to this if we're running non-interactively, where no prompt is shown.
+            if external_proc_exited && parser.is_interactive() {
+                let _ = std::io::stdout().flush();
```

Comment:
Seems reasonable.
Maybe we should do this only if the job is `signal_exited`?
I don't know if this scenario is also possible on normal exit.

Another approach is to do it in `Reader::readline()` before `exec_prompt`,
using something like `exec_external_count` to decide whether we want to flush.

(Both approaches won't handle the case when the output comes from a background job that's still running).

> Let me know if you think this should be included in the changelog afterwards.

Good question; can go both ways I think.
I don't think there is an obligation to add it

Reply:
I think keeping the flush call closer to the exec code makes more sense than putting it in the reader, btw.